### PR TITLE
releng: update pull-release-image-kube-cross to set the cluster where it should run

### PR DESCRIPTION
**Pull request by cpanato**
follow up of https://github.com/kubernetes/test-infra/pull/20689

the default cluster does not have the service account that is needed for this job, so adding the cluster that contains the setup

The error when the job tried to run

```
Pod can not be created: pods "63ae2683-6532-11eb-82fa-d61530c4f03a" ... count test-pods/gcb-builder: serviceaccount "gcb-builder" not found
```

/assign @saschagrunert @hasheddan @xmudrii @puerco 
cc: @kubernetes/release-engineering

### CODE REVIEW

**Comment by cpanato** [comment]
`jobs_test.go:351: pull-release-image-kube-cross: presubmits may not run in cluster: k8s-infra-prow-build-trusted`

/hold to see how to fix that

**Comment by cpanato** [comment]
@spiffxp @saschagrunert @hasheddan 

We need to trigger a GCB build for PRs to test the images, but looks like the `gcb-builder` service account is not available in pre-submits, I did not find another service account, does that exist? If not how can we add that?

**Review by hasheddan — COMMENTED** [review]
@cpanato I am guessing this is by design as typically we wouldn't be using this SA in a presubmit. However, in this specific case we have set up a testing repo so I would think the optimal scenario here would be to put a scoped token in the untrusted cluster if we want to maintain the idea of not running presubmits in trusted. Would defer to @spiffxp to weigh in though.

**Comment by cpanato** [comment]
@hasheddan agree with you on this, lets wait for Aaron to see other options

**Comment by spiffxp** [comment]
The reason the clusters are called "trusted" is that we don't want them running "untrusted" code, e.g. presubmits.  We treat the act of merging as the line of trust... code has passed tests, and the relevant humans have reviewed it.  So postubmits/periodics on trusted clusters are fine, but not presubmits.

@hasheddan's suggestion of scoped SA in untrusted cluster SGTM, PR to kubernetes/k8s.io would look something like:
- update scripts to make the GCP service account, bind via workload identity to an account of the same name on k8s-infra-prow-build (something like https://github.com/kubernetes/k8s.io/blob/master/infra/gcp/ensure-main-project.sh#L144-L155)
- give it privs to write to gcr for the project you care about (something like https://github.com/kubernetes/k8s.io/blob/35c51085f1096c01516021a8bcb34b9cfda656bd/infra/gcp/ensure-staging-storage.sh#L285-L295)
- add the service account resource to https://github.com/kubernetes/k8s.io/blob/master/infra/gcp/clusters/projects/k8s-infra-prow-build/prow-build/resources/build-serviceaccounts.yaml (eg: https://github.com/kubernetes/k8s.io/blob/master/infra/gcp/clusters/projects/k8s-infra-prow-build-trusted/prow-build-trusted/resources/build-serviceaccounts.yaml)

Now might be a good time to start adding some enforcement around which jobs are allowed to use which service accounts via tests in config/tests/jobs (ssuggested this in https://github.com/kubernetes/k8s.io/pull/1393#issuecomment-722002594)

This approach would also be baby steps toward "we should probably be using a service account per staging project" which might help address gcb quota issues https://github.com/kubernetes/test-infra/issues/20652

File: config/jobs/kubernetes/release/release-config.yaml
Comment (spiffxp) [inline config/jobs/kubernetes/release/release-config.yaml:87]: change this to `k8s-infra-prow-build`

change L93 to `serviceAccountName: gcb-builder-releng-test` (ref: https://github.com/kubernetes/k8s.io/pull/1626)

Comment (cpanato) [reply config/jobs/kubernetes/release/release-config.yaml]: thanks @spiffxp. updated

File: config/jobs/kubernetes/release/release-config.yaml
Comment (cpanato) [inline config/jobs/kubernetes/release/release-config.yaml:108]: the job will run on GCB so I don't think we need many resources

**Review by spiffxp — APPROVED** [review]
/lgtm
/hold cancel

**Pull request merged**
